test/test_autograd.py

```diff
@@ -1432,6 +1432,349 @@ def backward(self, dy):
             y = MyOp.apply(x)
         self.assertFalse(y.requires_grad)
 
+    def _is_inference_tensor(self, tensor):
```

Comment:
Maybe move all of these in a `TestAutogradInferenceMode` class to make it easier to separate from the rest.